src/vs/code/electron-main/window.ts

```diff
@@ -178,9 +174,9 @@ export class CodeWindow implements ICodeWindow {
 		}
 
 		let useCustomTitleStyle = false;
-		if (isMacintosh && (!windowConfig || !windowConfig.titleBarStyle || windowConfig.titleBarStyle === 'custom')) {
+		if ((isWindows || isMacintosh) && (!windowConfig || !windowConfig.titleBarStyle || windowConfig.titleBarStyle === 'custom')) {
 			const isDev = !this.environmentService.isBuilt || !!config.extensionDevelopmentPath;
-			if (!isDev) {
+			if (!isMacintosh || !isDev) {
```

Comment:
Why adding `!isMacintosh` here? Wouldn't that break the custom titlebar on Macs?

Reply:
I don't think it would. The `!isDev` addresses some bug (electron/electron#3647) that was fixed for Windows (for me at least). For MacOS, if `isDev` is false the custom titlebar will still work. The reason I added the `!isMacintosh` was to allow VS Code to use the custom titlebar in non-MacOS environments even when `isDev` is true (since there is no bug then).

Reply:
Why not putting that directly in the precedent if or straight in the variable declaration, like this:
```js
// not enabled when developing on Mac due to https://github.com/electron/electron/issues/3647
let useCustomTitleStyle = (!isMacintosh && windowConfig && windowConfig.titleBarStyle === 'custom') ||
                          (isMacintosh && (!windowConfig || !windowConfig.titleBarStyle || windowConfig.titleBarStyle === 'custom') && (!this.environmentService.isBuilt || !!config.extensionDevelopmentPath));
```

Might want to look also why is `!!` before `config.extensionDevelopmentPath` because the double `!` cancels themselves.

Reply:
Yeah it is a little weird to have that on the separate line. I was keeping the computation of `isDev` separate as it was before, which would make the if statement less complicated.

As for the `!!` it is my understanding that the operation casts `config.extensionDevelopmentPath` to a boolean. That way, if `!this.environmentService.isBuilt` is `false`, then `isDev` evaluates to `true` and not whatever the value of `config.extensionDevelopmentPath` is. However, given that this is plugged into a later `if` statement, this is likely unnecessary. It may only be so that TypeScript identifies `isDev` as a boolean.